### DISCUSSION TITLE
How to mimic Gmail inbox behavior?

### DISCUSSION THREAD

**Discussion by alexbraga** [TanStack/table#3630] [Q&A]
Hey guys,

So, to be clear on what I want, how can I make an entire row behave like a `<Link />` and render another component on click, while also preserving the action buttons at the end of it? My table has an `edit` and a `delete` button at each row, so the "clickable row" behavior must not override those buttons' actions.

Each row should be able to access a property called `movieId` from the fetched data, and then navigate to `/details/${movieId}`. I believe that it should be done where the table is rendered, probably where `row.getRowProps()` is called, but my knowledge of react-table is very limited yet, so I have no idea how to proceed.

Thanks in advance!

**Comment by Moshyfawn** [comment]
Look into [the `useRowSelect` hook](https://react-table.tanstack.com/docs/api/useRowSelect#userowselect) for RT. Might be what you need
